entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/PrimitiveFilteringExpressionConverter.java

```diff
@@ -55,10 +57,44 @@ public FilterTypeExpression convert(
     final String suffixedSubDocPath = identifierConverter.convert(metadata, requestContext);
 
     final IdentifierExpression identifierExpression = IdentifierExpression.of(suffixedSubDocPath);
-    final RelationalOperator relationalOperator = convertOperator(operator);
+    final RelationalOperator relationalOperator = normalizeOperator(operator, value);
     final ConstantExpression constantExpression =
         constantExpressionConverter.convert(constant, requestContext);
 
     return RelationalExpression.of(identifierExpression, relationalOperator, constantExpression);
   }
+
+  private RelationalOperator normalizeOperator(Operator operator, Value value)
+      throws ConversionException {
+    Operator normalizedOperator = operator;
+
+    // A null value filter translates to non-existence of attribute
```

Comment:
We never write nulls? I would think we'd want to translate `attr = null` to `(attr = null) OR (attr not exists)`, likewise on `!=`.

Reply:
How do you even write nulls via proto APIs? Unless the proto API implementation adds `null` value directly to the entities

That being said, it still makes sense to translate `attr = null` to `(attr = null) OR (attr not exists)` as you mentioned

Reply:
it could be written as string "null" for example - it depends on the calling code.